wolfcrypt/src/wc_port.c

```diff
@@ -423,6 +484,33 @@ void wc_ReadDirClose(ReadDirCtx* ctx)
 
 #endif /* !NO_FILESYSTEM && !NO_WOLFSSL_DIR */
 
+#if !defined(NO_FILESYSTEM) && defined(WOLFSSL_ZEPHYR)
+XFILE z_fs_open(const char* filename, const char* perm)
+{
+    XFILE file;
+
+    file = XMALLOC(sizeof(file), NULL, DYNAMIC_TYPE_FILE);
+    if (file != NULL)
+        fs_open(file, filename);
```

Comment:
Can fs_open() fail?  sizeof(file) is the right size for type type as well as the pointer to it?

Reply:
Yep, it can fail. Fixed.